fluent-syntax/src/parser.js

```diff
@@ -15,24 +28,36 @@ export function parse(source) {
   const entries = [];
 
   while (ps.current()) {
-    const entryStartPos = ps.getIndex();
+    const spanFrom = ps.getIndex();
 
     try {
       const entry = getEntry(ps);
-      if (entryStartPos === 0 && entry.type === 'Comment') {
+      if (spanFrom === 0 && entry.type === 'Comment') {
         comment = entry;
       } else {
+        entry.addSpan(spanFrom, ps.getIndex());
         entries.push(entry);
       }
-    } catch (e) {
-      entries.push(getJunkEntry(ps, source, entryStartPos));
-      errors.push(e);
+    } catch (err) {
+      ps.skipToNextEntryStart();
+
+      const spanTo = ps.getIndex();
+      const slice = source.substring(spanFrom, spanTo);
+      const junk = new AST.Junk(slice);
+      junk.addSpan(spanFrom, spanTo);
+      entries.push(junk);
+
+      errors.push(Object.assign(err, {
+        lineNumber: lineOffset(source, err.pos) + 1,
+        columnNumber: columnOffset(source, err.pos) + 1,
```

Comment:
`lineNumber` and `columnNumber` are modelled after [[1]] but I just realized that in `Errors`, `lineNumber` is 1-based and `columnNumber` is 0-based :/

Python is more consistent, having a 1-based `lineno` and a 0-based `col_offset` [[2]].

@Pike, what do you think would be the most useful for tools? Having worked a bit with ACE, for instance, I'd prefer to go for offsets both for lines and columns.

[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#Mozilla
[2]: https://docs.python.org/3/library/ast.html#ast.AST.lineno